dev-client/src/components/dataRequirements/handleMissingData.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useCallback} from 'react';
+import {ToastAndroid} from 'react-native';
+
+import {isFlagEnabled} from 'terraso-mobile-client/config/featureFlags';
+import {useNavigation} from 'terraso-mobile-client/navigation/hooks/useNavigation';
+
+export const useHandleMissingSite = () => {
+  const navigation = useNavigation();
+
+  return useCallback(() => {
+    navigation.navigate('BOTTOM_TABS');
```

Comment:
question: why do we use `navigation.navigate('BOTTOM_TABS')` here but `navigation.pop()` elsewhere?

Reply:
I was thinking because there are screens at various navigation depths that require sites, so instead of recursively pop()'ing, it'd be more straightforward to just always navigate to the same screen if we're missing a site.

I could change the navigation.pop() to navigate explicitly to the SiteNotesScreen though -- I don't think there's (currently) another screen that it'd pop() to in the case that a site note got deleted

Reply:
ohh like in a situation where you pop but the screen you popped to has also been deleted? do you know what would happen in this approach if we were just using `pop`? e.g. if we were in the `EditSiteNoteScreen` and the site got deleted, would it gracefully pop twice or would something weird happen?

i guess there's also a UX question here: if you navigate to a site from a project, and then that site gets deleted, should you be returned to the project or to the bottom tabs. but mainly i was just curious, cuz this is all pretty low stakes as long as it doesn't crash

Reply:
I didn't try it cuz I was like "this approach seems fine" 😅 so I'm not sure if we would do multiple pop()s gracefully or if something weird would happen. Since this case is fairly uncommon I lean towards "just do the easiest reasonable thing".

Yeah I think the question of "where should each screen navigate when each piece of data is missing" is a thing we will tackle as part of [#2290](https://github.com/techmatters/terraso-mobile-client/issues/2290) but also here am leaning towards "just do the easiest reasonable thing"